src/Documentation/Getting-Started-With-Orleans/Grain-Persistence.md

```diff
@@ -134,6 +134,9 @@ A quick way to test this is to (see in the aforementioned the few commented line
 The ADO.NET persistence has functionality to version data and define arbitrary (de)serializers with arbitrary application rules and streaming, but currently
 there is no method to expose them to application code. More information in [ADO.NET Persistence Rationale](#ADONETPersistenceRationale).
 
+__NB for Microsoft SQL Server users:__
+For the ADO.NET Storage Provider to work with SQL Server, you need to add [System.Data.SqlClient](https://www.nuget.org/packages/System.Data.SqlClient/) NuGet package to your project.
```

Comment:
About verbiage. Perhaps just as @shayhatsor outlines, something along these lines (don't use this as such :))
> Any database with ADO.NET provider is supported with any features set up in the provided scripts. At minimum the scripts support the membership provider (PostgreSql), SQL Server and MySQL also support membership, reminder and storage provider parts. The scripts can be found at the Nuget package (http://www.nuget.org/packages/Microsoft.Orleans.OrleansSqlUtils/, directory xyz). The ADO.NET providers corresponding the currently supported scripts are at https://www.nuget.org/packages/MySql.Data, https://www.nuget.org/packages/Npgsql/ and at https://www.nuget.org/packages/System.Data.SqlClient/.

Mentioning lists with links is helpful, but the trouble is naturally to keep them updated. It appears to be a bit so also there are newbies using relational DBs and they have inhererently more knobs and things to tune (especially in enterprise space) that might cause confusion. I would like to also mention other, more general thing, as to how could one setup a database and especially for the storage provider users on how to secure data, disaster recovery scenarios, hot/cold data (https://technet.microsoft.com/en-us/windows-server-docs/storage/storage-spaces/storage-spaces-direct-overview) perhaps and so on. A bit of a trouble is that I'm not currently working with this stuff, so I wouldn't like pretend offering "solid advice" in that regard. Perhaps it's changing, though. :)

About the ADO.NET issue... The PostgreSql tests are for membership only, since that's what's being used. The setup logic would need to overhauled a bit, since it's in two places -- storage and elsewhere -- and the setup is different in both of them. 

I would prefer to unify them and use constructor injection via DI throughout (doable also in XUnit.NET). There is a distinction there about setting up the test state invariants one normally sets class invariants, in a constructor, and then setting up environment, which I should be ensured is set up before the construction logic of any of the individual test classes -- maybe thinkinking so that the setup logic is shared amongst multiple test classes (and they may not need to share a common base class). I remember this distinction was a bit difficult to achieve with the approach of protected inheritance. The storage part went a bit into this direction, but I ran out of time to do it (indeed, the idea was also push a bit towards independent test kits). Anyhow, it's tangential to this discussion.